artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/proton/AMQPConnectionContext.java

```diff
@@ -469,7 +474,46 @@ public void onRemoteOpen(Connection connection) throws Exception {
       } catch (Exception e) {
          log.error("Error init connection", e);
       }
-      if (!validateConnection(connection)) {
+
+      boolean redirected = false;
+      if (connectionCallback.getTransportConnection().getRedirectTo() != null) {
+         org.apache.activemq.artemis.spi.core.remoting.Connection transportConnection = connectionCallback.getTransportConnection();
+         BrokerBalancer brokerBalancer = protocolManager.getServer().getBalancerManager()
+            .getBalancer(transportConnection.getRedirectTo());
+         Target target = brokerBalancer.getTarget(transportConnection,
+            connection.getRemoteContainer(), handler.getSASLResult().getUser());
+
+         if (target != null) {
+            ActiveMQServerLogger.LOGGER.redirectClientConnection(transportConnection, target);
+
+            if (!target.isLocal()) {
+               String host = ConfigurationHelper.getStringProperty(TransportConstants.HOST_PROP_NAME, TransportConstants.DEFAULT_HOST, target.getConnector().getParams());
+               int port = ConfigurationHelper.getIntProperty(TransportConstants.PORT_PROP_NAME, TransportConstants.DEFAULT_PORT, target.getConnector().getParams());
+
+               ErrorCondition error = new ErrorCondition();
+               error.setCondition(ConnectionError.REDIRECT);
+               error.setDescription(String.format("Connection redirected to %s by broker balancer %s", brokerBalancer.getName(), target));
```

Comment:
Are the args possibly reversed? Havent tried it, but the format string seems to read more like the balancer name is expected to be second/last?

Either way, the toString() on Target seems like this might result in including a bunch of config/names some might consider 'internal' to the broker setup into the description. Or perhaps just not entirely relevant to an AMQP client/user that doesnt use them. 

It might be simpler and clearer just to use the target host:port, which are already available here.

Reply:
yes the args are reversed. I'll replace the target.toString() with host:port